// @instructure/ui-themes/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "additionalPrimitives", {
  enumerable: true,
  get: function () {
    return _primitives.additionalPrimitives;
  }
});
Object.defineProperty(exports, "canvas", {
  enumerable: true,
  get: function () {
    return _canvas.default;
  }
});
Object.defineProperty(exports, "canvasHighContrast", {
  enumerable: true,
  get: function () {
    return _canvasHighContrast.default;
  }
});
Object.defineProperty(exports, "canvasHighContrastThemeLocal", {
  enumerable: true,
  get: function () {
    return _canvasHighContrast.canvasHighContrastThemeLocal;
  }
});
Object.defineProperty(exports, "canvasThemeLocal", {
  enumerable: true,
  get: function () {
    return _canvas.canvasThemeLocal;
  }
});
Object.defineProperty(exports, "dataVisualization", {
  enumerable: true,
  get: function () {
    return _dataVisualization.default;
  }
});
exports.default = void 0;
Object.defineProperty(exports, "primitives", {
  enumerable: true,
  get: function () {
    return _primitives.primitives;
  }
});
var _canvasHighContrast = _interopRequireWildcard(require("./themes/canvasHighContrast"));
var _canvas = _interopRequireWildcard(require("./themes/canvas"));
var _primitives = require("./sharedThemeTokens/colors/primitives");
var _dataVisualization = _interopRequireDefault(require("./sharedThemeTokens/colors/dataVisualization"));
/*
 * The MIT License (MIT)
 *
 * Copyright (c) 2015 - present Instructure, Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
var _default = exports.default = _canvas.default;
