// @instructure/emotion/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  InstUISettingsProvider: true,
  withStyle: true,
  ThemeablePropValues: true,
  ThemeablePropTypes: true,
  makeThemeVars: true,
  getShorthandPropValue: true,
  mirrorShorthandCorners: true,
  mirrorShorthandEdges: true,
  mapSpacingToShorthand: true,
  useStyle: true
};
Object.defineProperty(exports, "InstUISettingsProvider", {
  enumerable: true,
  get: function () {
    return _InstUISettingsProvider.InstUISettingsProvider;
  }
});
Object.defineProperty(exports, "ThemeablePropTypes", {
  enumerable: true,
  get: function () {
    return _styleUtils.ThemeablePropTypes;
  }
});
Object.defineProperty(exports, "ThemeablePropValues", {
  enumerable: true,
  get: function () {
    return _styleUtils.ThemeablePropValues;
  }
});
Object.defineProperty(exports, "getShorthandPropValue", {
  enumerable: true,
  get: function () {
    return _styleUtils.getShorthandPropValue;
  }
});
Object.defineProperty(exports, "makeThemeVars", {
  enumerable: true,
  get: function () {
    return _styleUtils.makeThemeVars;
  }
});
Object.defineProperty(exports, "mapSpacingToShorthand", {
  enumerable: true,
  get: function () {
    return _styleUtils.mapSpacingToShorthand;
  }
});
Object.defineProperty(exports, "mirrorShorthandCorners", {
  enumerable: true,
  get: function () {
    return _styleUtils.mirrorShorthandCorners;
  }
});
Object.defineProperty(exports, "mirrorShorthandEdges", {
  enumerable: true,
  get: function () {
    return _styleUtils.mirrorShorthandEdges;
  }
});
Object.defineProperty(exports, "useStyle", {
  enumerable: true,
  get: function () {
    return _useStyle.useStyle;
  }
});
Object.defineProperty(exports, "withStyle", {
  enumerable: true,
  get: function () {
    return _withStyle.withStyle;
  }
});
var _react = require("@emotion/react");
Object.keys(_react).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _react[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _react[key];
    }
  });
});
var _InstUISettingsProvider = require("./InstUISettingsProvider");
var _withStyle = require("./withStyle");
var _styleUtils = require("./styleUtils");
var _useStyle = require("./useStyle");
