// @instructure/emotion/lib/InstUISettingsProvider/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InstUISettingsProvider = InstUISettingsProvider;
exports.default = void 0;
var _react = require("react");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react2 = require("@emotion/react");
var _TextDirectionContext = require("@instructure/ui-i18n/lib/TextDirectionContext.js");
var _DeterministicIdContextProvider = require("@instructure/ui-react-utils/lib/DeterministicIdContext/DeterministicIdContextProvider.js");
var _getTheme = require("../getTheme");
var _jsxRuntime = require("@emotion/react/jsx-runtime");
/*
 * The MIT License (MIT)
 *
 * Copyright (c) 2015 - present Instructure, Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

/**
 * ---
 * category: components/utilities
 * ---
 * @module InstUISettingsProvider
 */
function InstUISettingsProvider({
  children,
  theme = {},
  dir,
  instanceCounterMap,
  as
}) {
  const finalDir = dir || (0, _react.useContext)(_TextDirectionContext.TextDirectionContext);
  if (process.env.NODE_ENV !== 'production' && finalDir === 'auto') {}
  let providers = (0, _jsxRuntime.jsx)(_DeterministicIdContextProvider.DeterministicIdContextProvider, {
    instanceCounterMap: instanceCounterMap,
    children: (0, _jsxRuntime.jsx)(_react2.ThemeProvider, {
      theme: (0, _getTheme.getTheme)(theme),
      children: (0, _jsxRuntime.jsx)(_TextDirectionContext.TextDirectionContext.Provider, {
        value: finalDir,
        children: children
      })
    })
  });
  if (dir) {
    const Element = as || 'span';
    providers = (0, _jsxRuntime.jsx)(Element, {
      dir: finalDir,
      children: providers
    });
  } else if (as && process.env.NODE_ENV !== 'production') {}
  return providers;
}
InstUISettingsProvider.propTypes = {
  /* eslint-disable react/require-default-props */
  children: _propTypes.default.node,
  theme: _propTypes.default.oneOfType([_propTypes.default.object, _propTypes.default.func]),
  dir: _propTypes.default.oneOf(['ltr', 'rtl']),
  instanceCounterMap: _propTypes.default.instanceOf(Map),
  as: _propTypes.default.string
  /* eslint-enable react/require-default-props */
};
var _default = exports.default = InstUISettingsProvider;
